axelrod/tests/strategies/test_memorytwo.py

```diff
@@ -1,11 +1,193 @@
+                          ((D, D), (D, C)): 1.0, ((D, D), (D, D)): 1.0})
+
+    def test_exception_if_four_vector_not_set(self):
+        with warnings.catch_warnings(record=True) as warning:
+            warnings.simplefilter("always")
+            player = MemoryTwoPlayer()
+
+            self.assertEqual(len(warning), 1)
+            self.assertEqual(warning[-1].category, UserWarning)
+            self.assertEqual(str(warning[-1].message),
+                              "Memory two player is set to default, Cooperator.")
+
+    def test_exception_if_probability_vector_outside_valid_values(self):
+        player = MemoryTwoPlayer()
+        x = 2
+        with self.assertRaises(ValueError):
+            player.set_sixteen_vector([0.1, x, 0.5, 0.1, 0.1, 0.2,
+                                       0.5, 0.1, 0.1, 0.2, 0.5, 0.1,
+                                       0.2, 0.5, 0.1, 0.2, 0.5, 0.2,])
+
+
+class TestMemoryStochastic(TestPlayer):
+    name = 'Generic Memory Two Player: (1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1): C'
+    player = axelrod.MemoryTwoPlayer
+    expected_classifier = {
+        'memory_depth': 2,  # Memory-two Sixteen-Vector
+        'stochastic': False,
+        'makes_use_of': set(),
+        'long_run_time': False,
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+    def test_strategy(self):
+        axelrod.seed(0)
+        vector = [random.random() for _ in range(16)]
+
+        actions = [(C, C), (C, C), (D, D), (D, C), (C, C), (C, D), (C, C)]
+        self.versus_test(opponent=axelrod.CyclerCCD(),
+                         expected_actions=actions, seed=0,
+                         init_kwargs={'sixteen_vector': vector})
+
+        actions = [(C, C), (C, C), (C, D), (D, C), (C, C), (C, D), (C, C)]
+        self.versus_test(opponent=axelrod.CyclerCCD(),
+                         expected_actions=actions, seed=1,
+                         init_kwargs={'sixteen_vector': vector})
+
+        actions = [(C, C), (C, C), (D, C), (D, D), (C, D), (C, C), (D, C)]
+        self.versus_test(opponent=axelrod.TitForTat(),
+                         expected_actions=actions, seed=0,
+                         init_kwargs={'sixteen_vector': vector})
+
+        actions = [(C, C), (C, C), (C, C), (D, C), (D, D), (C, D), (C, C)]
+        self.versus_test(opponent=axelrod.TitForTat(),
+                         expected_actions=actions, seed=1,
+                         init_kwargs={'sixteen_vector': vector})
+
+
+class TestAON2(TestPlayer):
+
+    name = "AON2"
+    player = axelrod.AON2
+    expected_classifier = {
+        'memory_depth': 2,
+        'stochastic': False,
+        'makes_use_of': set(),
+        'long_run_time': False,
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+
+    def test_class_classification(self):
```

Comment:
Does this test have to be explicit? Or is it covered by the generic player tests?